src/reducers/arcReducer.ts

```diff
@@ -466,11 +466,27 @@ const arcReducer = (state = initialState, action: any) => {
       switch (sequence) {
         case AsyncActionSequence.Pending:
           return update(state, {
-            redemptions: { [redemptionsKey] : { transactionState: { $set: TransactionStates.Unconfirmed }} }
+            redemptions: {
+              [redemptionsKey]: {
+                $apply: (redemptions: IRedemptionState) => ({
+                  ...redemptions,
```

Comment:
why do we need to update the redemption here?

Reply:
I was getting `cannot read transactionState of undefined` because immutability-helper tries to read it from an undefined redemption. 
This simply does the same thing but avoids this error.

Reply:
well yeah, but that error should not be happening right? the redemption should be there. I guess this is a decent failsafe, but I worry that it could hide another error then. can you think of a valid reason why the redemption wouldn't be there?

Reply:
I think only `transactionState` is not defined on the redemption, need to check. Though it makes sense that it's undefined until the first redeem. Or you can have an `idle` state if you want.